.github/workflows/release-update-stable-tag.yml

```diff
@@ -0,0 +1,192 @@
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version (in X.Y.Z or X.Y format)'
+        required: true
+        default: ''
+      dry-run:
+        description: 'Do a dry-run of the stable tag update (if UNCHECKED, it will commit the change!)'
+        type: boolean
+        required: false
+        default: true
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  validate-release-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.extract-version.outputs.version }}
+      branch: ${{ steps.extract-version.outputs.branch }}
+    steps:
+      - name: Validate release version
+        id: extract-version
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const versionInput = '${{ github.event.inputs.version }}';
+
+            // Match X.Y.Z or X.Y formats
+            const versionMatch = versionInput.match(/^(\d+\.\d+)(?:\.(\d+))?$/);
+
+            if (versionMatch) {
+              const major = versionMatch[1]; // X.Y
+              const patch = versionMatch[2] || '0'; // Z (default to 0 if not provided)
+
+              const version = `${major}.${patch}`;
+              const branch = `release/${major}`;
+
+              console.log(`Input version: ${versionInput}`);
+              console.log(`Normalized version: ${version}`);
+              console.log(`Release branch: ${branch}`);
+              core.setOutput('version', version);
+              core.setOutput('branch', branch);
+            } else {
+              core.setFailed(`Version ${versionInput} does not match the expected format (X.Y.Z or X.Y)`);
+              process.exit(1);
+            }
+
+  verify-release-branch:
+    runs-on: ubuntu-latest
+    needs: validate-release-version
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Verify release branch exists
+        run: |
+          RELEASE_BRANCH="${{ needs.validate-release-version.outputs.branch }}"
+          if ! git ls-remote --heads origin "$RELEASE_BRANCH" | grep -q "refs/heads/$RELEASE_BRANCH"; then
+            echo "❌ ERROR: Release branch '$RELEASE_BRANCH' does not exist"
+            exit 1
+          else
+            echo "✅ Release branch '$RELEASE_BRANCH' exists"
+          fi
+
+  update-wporg-stable-tag:
+    runs-on: ubuntu-latest
+    needs: [validate-release-version, verify-release-branch]
+    steps:
+      - name: Install SVN
+        run: sudo apt-get update && sudo apt-get install -y subversion
+
+      - name: Update SVN stable tag
+        env:
+          TARGET_VERSION: ${{ needs.validate-release-version.outputs.version }}
+        run: |
+          echo "🔄 Checking out SVN repository..."
+          svn checkout "${{ secrets.SVN_REPO_URL }}" ./svn-woocommerce-repo --username "${{ secrets.SVN_USERNAME }}" --password "${{ secrets.SVN_PASSWORD }}" --non-interactive
```

Comment:
We can use `svn checkout --depth` (see https://svnbook.red-bean.com/en/1.6/svn.advanced.sparsedirs.html) to only retrieve `trunk` or even just the readme.txt. It's not necessary to checkout the full repo.